Clarify shell restarting message

Gurkenglas opened the pull request:
Addresses https://github.com/rust-lang/rustup/issues/2681.

Gurkenglas committed:
Clarify `post_install_msg`s advice

Someone who supplies a `no_modify_path` option requires less clarification.

Gurkenglas committed:
Clarify FAQ

rbtcollins commented on src/cli/self_update.rs:
```diff
@@ -162,9 +162,9 @@ macro_rules! post_install_msg_unix {
-To get started you need Cargo's bin directory ({cargo_home}/bin) in your `PATH`
-environment variable. Next time you log in this will be done
-automatically.
+To get started you may need to log out and back in.
```
Perhaps.  ...out and back in, or start a new shell. (We've changed what we do on unix since these message was written, and any new shell will now affect PATH.

rbtcollins reviewed (changes requested):
One small tweak requested :)

Gurkenglas committed:
Relog advice is now shell restart advice

Restarting the shell is now enough. Relogging restarts the shell so we don't need to suggest relogging.
